### PR TITLE
Use errors.Join

### LINKED ISSUE #216: Replace github.com/joeshaw/multierror with errors.Join
We could probably replace `github.com/joeshaw/multierror` with the stdlib https://pkg.go.dev/errors#Join (added in go1.20).

### PR DESCRIPTION
**Pull request by andrewkroh**
Replace usages of github.com/joeshaw/multierror with `errors.Join`, which was added to the stdlib in Go 1.20.

Closes #216

### CODE REVIEW

File: providers/linux/os.go [hunk 1]
```diff
@@ -229,7 +228,7 @@ func findDistribRelease(baseDir string) (*types.OSInfo, error) {
 		}
 		return osInfo, err
 	}
-	return nil, fmt.Errorf("no valid /etc/<distrib>-release file found: %w", &multierror.MultiError{Errors: errs})
+	return nil, fmt.Errorf("no valid /etc/<distrib>-release file found: %w", errors.Join(errs...))
```
Comment (efd6) [inline providers/linux/os.go:231]: To clarify that this is an error path, perhaps make the return at L229 be `return osInfo, nil`?

**Review by efd6 — DISMISSED** [review]
Nit, but LGTM

File: providers/linux/os.go (hunk 1 above)
Comment (andrewkroh) [reply providers/linux/os.go:231]: Yeah, that looks better. 4b71d5d

**Comment by andrewkroh** [comment]
Apparently this repo has a setting to clear approvals if there are new commits.

**Review by efd6 — APPROVED** [review]

**Pull request merged**